### PR TITLE
bite-1408 Add validation and fix restart bug

### PR DESCRIPTION
**Pull request by devlinmr**

### CODE REVIEW

**Comment by paulschwarzenberger** [comment]
Suggest combining with changes from PR for BITE 1396 including specific version of nginx

File: nginx-ingress-vault/nginx/virtual_host.go [hunk 1]
```diff
@@ -157,3 +159,34 @@ func newHTTPClient(dest *url.URL) *http.Client {
     return &http.Client{}
 }
 
+func (vhost *VirtualHost) ValidateVirtualHost() error {
```
Comment (geriBatai) [inline nginx-ingress-vault/nginx/virtual_host.go]: I'd just call it Validate(), too much repetition on this line :)

File: nginx-ingress-vault/nginx/virtual_host.go [hunk 2]
```diff
@@ -157,3 +159,34 @@ func newHTTPClient(dest *url.URL) *http.Client {
...
+
+    schemeRegex, _ := regexp.Compile("^https?$")
+    hostRegex, _ := regexp.Compile("[a-z\\d+].*?\\.\\w{2,8}$")
+
+    if reflect.TypeOf(vhost.Name).String() != "string" || vhost.Name == "" {
```
Comment (cristian-radu) [inline nginx-ingress-vault/nginx/virtual_host.go]: Doesn't kube api already check that you have a name set for the ingress ?

**Comment by paulschwarzenberger** [comment]
need to update branch to include version number 1.9.0 in controller.go

File: nginx-ingress-vault/nginx/virtual_host.go (hunk 1 above)
Comment (devlinmr) [inline nginx-ingress-vault/nginx/virtual_host.go]: Yeh but we got a couple of validates for other things

File: nginx-ingress-vault/nginx/virtual_host.go (hunk 2 above)
Comment (devlinmr) [inline nginx-ingress-vault/nginx/virtual_host.go]: To be sure, to be sure

File: nginx-ingress-vault/nginx/virtual_host.go (hunk 1 above)
Comment (geriBatai) [inline nginx-ingress-vault/nginx/virtual_host.go]: (vhost *VirtualHost) shows what this Validate() will be performed against.

Comment (devlinmr) [inline nginx-ingress-vault/nginx/virtual_host.go]: Renamed it to Validate

**Comment by paulschwarzenberger** [comment]
tested successfully, confirmed acccess to lb-prelive, and checked ingress logs.
LGTM

**Comment by bsomogyi** [comment]
LGTM

**Pull request merged**